core/variable.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2012 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Components for the variable model.
+ * @author marisaleung@google.com (Marisa Leung)
+ */
+'use strict';
+
+goog.provide('Blockly.Variable');
+
+/**
+ * Class for a variable.
+ * @param {!string} name The name of the variable.
+ * @param {!string} type The type of the variable.
+ * @param {!string} id The unique id of the variable.
+ * @constructor
+ */
+Blockly.Variable = function(name, type, id) {
+  /*
+   * @type {!string}
+   * 
+   * The name of the variable, typically defined by the user. It must be
+   * unique across all names used for procedures and variables. It may be
+   * changed by the user.
+   */
+  this.name = name;
+
+  /*
+   * @type {!string}
+   * 
+   * A unique id for the variable. This should be defined at creation and
+   * not change, even if the name changes. In most cases this should be a
+   * UUID.
+   */
+  this.id_ = id;
+
+  /*
+   * @type {!string}
+   * 
+   * The type of the variable. By default this is the empty string '' which
```

Comment:
Include examples to make it clearer what type is in this case. Something like,
"The type of the variable, such as 'int' or 'sound_effect'. This may be used to build a list of variables of a specific type. By default ..."